loadtests/loadtest/tutorial.py

```diff
@@ -0,0 +1,308 @@
+        resp = self.session.get(
+            self.collection_url('default', collection_id),
+            auth=self.auth,
+            headers={'Content-Type': 'application/json'})
+        self.incr_counter("status-%s" % resp.status_code)
+        self.assertEqual(resp.status_code, 200)
+        records = resp.json()['data']
+        self.assertEqual(len(records), 2)
+        etag = resp.headers['ETag']
+
+        # Fetch the collection from the Etag and see nothing new
+        url = self.collection_url('default', collection_id)
+        resp = self.session.get(url,
+                                params={'_since': etag.strip('"')},
+                                auth=self.auth,
+                                headers={'Content-Type': 'application/json'})
+        self.incr_counter("status-%s" % resp.status_code)
+        self.assertEqual(resp.status_code, 200)
+        records = resp.json()['data']
+        self.assertEqual(len(records), 0)
+
+        # Update a task
+        resp = self.session.patch(
+            self.record_url(record_id, 'default', collection_id),
+            data=json.dumps({'data': {'status': 'done'}}),
+            auth=self.auth,
+            headers={'Content-Type': 'application/json'})
+        self.incr_counter("status-%s" % resp.status_code)
+        self.assertEqual(resp.status_code, 200)
+        record = resp.json()
+        self.assertEqual(record['data']['status'], 'done')
+        self.assertIn('write', record['permissions'])
+
+        # Try an update with If-Match on the saved ETag and see it fails
+        resp = self.session.patch(
+            self.record_url(record_id, 'default', collection_id),
+            data=json.dumps({'data': {'status': 'done'}}),
+            auth=self.auth,
+            headers={'Content-Type': 'application/json',
+                     'If-Match': etag})
+        self.incr_counter("status-%s" % resp.status_code)
+        self.assertEqual(resp.status_code, 412)
+        self.assertEqual(resp.headers['ETag'],
+                         '"%s"' % record['data']['last_modified'])
+
+        # Get the list of records and update the ETag
+        resp = self.session.get(
+            '%s?_since=%s' % (self.collection_url('default', collection_id),
+                              etag.strip('"')),
+            auth=self.auth,
+            headers={'Content-Type': 'application/json'})
+        self.incr_counter("status-%s" % resp.status_code)
+        self.assertEqual(resp.status_code, 200)
+        records = resp.json()['data']
+        self.assertEqual(len(records), 1)
+        etag = resp.headers['ETag']
+
+        # Try an update with If-Match on the new ETag and see it works
+        resp = self.session.patch(
+            self.record_url(record_id, 'default', collection_id),
+            data=json.dumps({'data': {'status': 'done'}}),
+            auth=self.auth,
+            headers={'Content-Type': 'application/json',
```

Comment:
I think you can set this header on the session object instead of repeating it at each request